File: Source/Tuple.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ProgCom
{
    public class Tuple<T,U>
    {
        public T Item1;
        public U Item2;
        public Tuple(T item1, U item2)
        {
            this.Item1 = item1;
            this.Item2 = item2;
        }
    }
}

File: Compiler/source/Tuple.java

public class Tuple<A, B>{
	public final A x;
	public final B y;
	public Tuple(A X, B Y) {
		x = X;
		y = Y;
	}
	public String toString() {
		return "["+x.toString()+":"+y.toString()+"]";
		
	}
}
